File: 프로그래머스/H-Index.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<int> citations) {
    int answer = 0;
    int h=0;
    bool check=false;
    sort(citations.begin(),citations.end());
    reverse(citations.begin(),citations.end());
    
    for(int i=0;i<=citations.size();i++){
        
        if(i>=citations[i]){
            answer=i;
            break;
        }
    }
    
    
    return answer;
}

File: 프로그래머스/Lv2/H-Index.py
def solution(citations):
    answer = 0
    citations.sort()
    Size=len(citations)
    
    for i in range(1,Size+1):
        for j in range(Size):
            if(citations[j]>=i): #h번 이상 인용된 논문일때
                if(Size-j>=i and j<=i):
                    answer=i
    
    return answer
